### PR TITLE
Move from Travis to GitHub Actions

### PR DESCRIPTION
**Pull request by cmharlow**
JIRA ticket: https://jira.mozilla.com/browse/SE-1710

tldr; - Travis.org, which this repo uses / used, is being migrated to Travis.com, so Travis pipeline runs for this repo are sporadic / inconsistent. We could move to Travis.com, but Travis.com for mozilla GH org is currently deprecated (see convo in Slack about this). So, this PR moves from Travis to GitHub actions.

things to note:
* this will run flake8, eslint, stylelint on all commits (pushes / merges to branches) & creation of PRs
* this will also build and push a Docker image (tagged with the short commit) upon all commits, merges, creation of PRs
* when you update python versions, we'll need to update that in the GHA configurations as well
* added some clear steps to our working docs on how to fully deploy this app and handle some shortcuts as needed

open question: if you'd like to compare or go with CircleCI, let me know and that can be pushed as well.

### CODE REVIEW

**Comment by sciurus** [comment]
@cmharlow I'll defer to devs on what CI they prefer, since they are the ones who will own maintaining this.

My only question is does the tagging scheme here reflect what flux is looking for to determine whether to deploy to stage vs prod.

**Comment by cmharlow** [comment]
@sciurus so flux doesn’t look for any docker image tags in this setup - through kustomize, is applies whatever files are in kubernetes/{env}, no patches.

However, this setup does change the image tagging a bit. It uses the shorter SHA, and will tag with the GH release tag if such is created (though I don’t believe this repository uses tagging, and no deployments are driven by docker tags alone here, so it may be a moot addition).

**Comment by sciurus** [comment]
AH, got it. Thanks for digging in and clarifying!

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 2.7
+      uses: actions/setup-python@v2
+      with:
+        python-version: 2.7.18
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade 'pip==20.3.1'
+        python -m pip install flake8
```
Comment (akatsoulas) [inline .github/workflows/ci.yaml:18]: flake8 should already be in requirements.dev so we could remove this step

**Review by akatsoulas — APPROVED** [review]
@cmharlow this looks good to me. Just added a couple of nits

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        python -m pip install -r requirements/dev.txt
+    - name: Lint with flake8
+      run: flake8 moderator --count --show-source --statistics
+
+  eslint-stylelint:
```
[the remark this answers is not in the document]
Comment (cmharlow) [reply .github/workflows/ci.yaml:23]: cool, will do

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (cmharlow) [reply .github/workflows/ci.yaml:18]: great, thanks! love cleaning up duplications

File: .github/workflows/ci.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (cmharlow) [reply .github/workflows/ci.yaml]: removed it

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (cmharlow) [reply .github/workflows/ci.yaml]: removed the flake8 install line
